packages/tablev2/Table.tsx

```diff
@@ -0,0 +1,306 @@
+import * as React from "react";
+import { cx } from "emotion";
+import * as style from "./style";
+import { Draggable, HeaderCellContent, Sorter } from "./Util";
+import { ResetButton } from "../button";
+
+export type Column<A, B = string> = {
+  /**
+   * An id given to that column. Needed for persistence.
+   */
+  id: B;
+
+  /**
+   * Renderer for the cells of that column.
+   */
+  render: (a: A) => React.ReactNode;
+
+  /**
+   * Render a header cell
+   */
+  header?: React.ReactNode;
+
+  /**
+   * Helper content that explains the content in the column.
+   */
+  headerTooltipContent?: React.ReactNode;
+
+  /**
+   * A function to sort the entries of the table.
+   * It first needs to handle a direction and then has to compare two elements.
+   * e.g. (dir: "asc" | "desc", dirAsNum: 1 | -1) => (a: number, b: number ) => dirAsNum * (a - b)
+   */
+  sorter?: Sorter<A>;
+
+  /**
+   * The initial width of that column. Will be interpolated to a grid-template:
+   * https://developer.mozilla.org/en-US/docs/Web/CSS/grid-template-columns
+   * Fine:
+   * - `"20px"`
+   * - `"10%"`
+   * - `"1fr"`
+   * - `"mixmax(100px, 1fr)"`
+   * - ...
+   * Note that each row calculates its own grid, so using e.g max-content might not be a good idea.
+   */
+  initialWidth?: string;
+
+  /**
+   * Text alignment for cells in the column
+   */
+  textAlign?: React.CSSProperties["textAlign"];
+};
+
+// TODO: make it possible to disable rows from being selected with a checkbox
+type TableProps<A, B = string, C extends B = B> = {
+  /**
+   * Table columns
+   */
+  columns: Array<Column<A, B>>;
+
+  /**
+   * The ID and direction to initially sort by.
+   */
+  initialSorter?: { by: C; order?: "asc" | "desc" };
+
+  /**
+   * A callback invoked whenever the internal state of the table is changed.
+   * Have a look at the State type to see what's in there.
+   */
+  onStateChange?: (a: State) => void;
+
+  /**
+   * Whether the first column shall be sticky. Defaults to true.
+   */
+  stickyFirstCol?: boolean;
+
+  /**
+   * A function that provides a stable and unique ID for entries in the table.
+   */
+  toId: (a: A) => string;
+};
+
+// DON'T put stuff here that you don't want persisted.
+type State = {
+  sortBy: string | null;
+  order: "asc" | "desc";
+  columns: Array<{
+    id: string;
+    width: string;
+  }>;
+};
+
+const getCellClassName = (
+  gridTemplateFragments: string[],
+  stickyFirstCol?: boolean
+) =>
+  cx(style.row(gridTemplateFragments), {
+    [style.rowWithStickyColumn]: stickyFirstCol
+  });
+
+type HeaderCellProps<Entry> = {
+  column: Column<Entry>;
+  selectableRowIds?: string[];
+  id: string;
+  isFirstColumn: boolean;
```

Comment:
Why do we need this?

Reply:
we don't. it's a leftover from the selectableRow-functionality.